lib/kontena/plugin/cloud/image/login_command.rb

```diff
@@ -0,0 +1,59 @@
+require_relative 'common'
+require 'shellwords'
+require 'open3'
+
+class Kontena::Plugin::Cloud::Image::LoginCommand < Kontena::Command
+  include Kontena::Cli::Common
+  include Kontena::Plugin::Cloud::Image::Common
+
+  requires_current_account_token
+
+  def execute
+    token = nil
+    spinner "Creating a new Kontena Cloud token for Docker" do
+      data = { attributes: { name: 'docker' } }
+      token = cloud_client.post("/user/personal_access_tokens", { data: data })['data']
+    end
+
+    success = spinner "Logging in to the Kontena Cloud Image Registry" do |spin|
+      pass = token.dig('attributes', 'access-token')
+
+      if `docker login --help`['--password-stdin']
+        output, stderr, status = Open3.capture3("docker login -u %s --password-stdin %s" % [current_account.username, image_distribution_url].map(&:shellescape), :stdin_data => pass)
+        unless status.success?
+          exit_with_error stderr
+        end
+      else
+        pass = 'foo'
```

Comment:
Does not work if someone's password is not "foo"

Reply:
ah, some left-over from failure injection testing 🤦‍♂️